### PR TITLE
update py3dns

### PR DESCRIPTION
**Pull request by ethorsoe**
Bump py3dns version to make is import without asyncore on python-3.12
https://launchpad.net/py3dns/+milestone/4.0.0

rename attribute pydns to py3dns to match the package name

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ethorsoe** [comment]
tested with (apart from nixpkgs-review)

```
$ NIX_PATH=nixpkgs=/tmp/nixpkgs/ nix-shell -p 'python312.withPackages (p: [ p.py3dns ])' --command python3.12
install: cannot remove '/tmp/env-vars': Operation not permitted
Python 3.12.6 (main, Sep  6 2024, 19:03:47) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from DNS import ParseResolvConf, Request
>>> ParseResolvConf()
>>> Request(qtype='A').req("nixos.org").answers
[{'name': 'nixos.org', 'type': 1, 'class': 1, 'ttl': 20, 'rdlength': 4, 'typename': 'A', 'classstr': 'IN', 'data': '3.72.140.173'}, {'name': 'nixos.org', 'type': 1, 'class': 1, 'ttl': 20, 'rdlength': 4, 'typename': 'A', 'classstr': 'IN', 'data': '52.58.254.253'}]
```

File: pkgs/development/python-modules/py3dns/default.nix [hunk 1]
```diff
@@ -2,24 +2,20 @@
...
   fetchPypi,
+  flit-core,
 }:
 
 buildPythonPackage rec {
   pname = "py3dns";
-  version = "3.2.1";
-  format = "setuptools";
+  version = "4.0.2";
+  format = "pyproject";
```
Comment (mweinelt) [inline pkgs/development/python-modules/py3dns/default.nix:11]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/py3dns/default.nix [hunk 2]
```diff
@@ -2,24 +2,20 @@
...
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1r25f0ys5p37bhld7m7n4gb0lrysaym3w318w2f8bncq7r3d81qz";
+    sha256 = "sha256-mGUugOzsFDxg948OazQWMcqadWDt2N3fyGTAKQJhijk=";
```
Comment (mweinelt) [inline pkgs/development/python-modules/py3dns/default.nix:15]:
```suggestion
    hash = "sha256-mGUugOzsFDxg948OazQWMcqadWDt2N3fyGTAKQJhijk=";
```

File: pkgs/development/python-modules/py3dns/default.nix [hunk 3]
```diff
@@ -2,24 +2,20 @@
...
   };
 
-  preConfigure = ''
-    sed -i \
-      -e '/import DNS/d' \
-      -e 's/DNS.__version__/"${version}"/g' \
-      setup.py
-  '';
+  nativeBuildInputs = [ flit-core ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/py3dns/default.nix:18]:
```suggestion
  build-system = [ flit-core ];
```

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -11224,7 +11224,7 @@ self: super: with self; {
 
   pydmd = callPackage ../development/python-modules/pydmd { };
 
-  pydns = callPackage ../development/python-modules/py3dns { };
+  py3dns = callPackage ../development/python-modules/py3dns { };
```
Comment (mweinelt) [inline pkgs/top-level/python-packages.nix:11227]: Add an alias for the old name to python-aliases.nix

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (ethorsoe) [reply pkgs/top-level/python-packages.nix:11227]: done to the best of my ability.

**Comment by kirillrdy** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351242`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.py3dns</li>
    <li>python311Packages.py3dns.dist</li>
    <li>python311Packages.pymilter</li>
    <li>python311Packages.pymilter.dist</li>
    <li>python311Packages.pyspf</li>
    <li>python311Packages.pyspf.dist</li>
    <li>python312Packages.py3dns</li>
    <li>python312Packages.py3dns.dist</li>
    <li>python312Packages.pymilter</li>
    <li>python312Packages.pymilter.dist</li>
    <li>python312Packages.pyspf</li>
    <li>python312Packages.pyspf.dist</li>
    <li>spf-engine</li>
    <li>spf-engine.dist</li>
    <li>trustymail</li>
    <li>trustymail.dist</li>
  </ul>
</details>

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Thank you!

**Pull request merged**